llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -9571,15 +9571,14 @@ getConstantEvolvingPHIOperands(Instruction *UseInst, const Loop *L,
     if (!OpInst || !canConstantEvolve(OpInst, L)) return nullptr;
 
     PHINode *P = dyn_cast<PHINode>(OpInst);
-    if (!P)
+    if (!P) {
       // If this operand is already visited, reuse the prior result.
       // We may have P != PHI if this is the deepest point at which the
       // inconsistent paths meet.
-      P = PHIMap.lookup(OpInst);
-    if (!P) {
       // Recurse and memoize the results, whether a phi is found or not.
       // This recursive call invalidates pointers into PHIMap.
-      P = getConstantEvolvingPHIOperands(OpInst, L, PHIMap, Depth + 1);
+      P = PHIMap.lookup_or(
+          OpInst, getConstantEvolvingPHIOperands(OpInst, L, PHIMap, Depth + 1));
```

Comment:
This is goign to unconditionally recurse to compute the default. The point of the check is to not do the recursion if the entry already exists...